### PR TITLE
Changes required to fix autorest.typescript tests

### PR DESCRIPTION
**Pull request by daschult**

### CODE REVIEW

**Comment by RikkiGibson** [comment]
Never change, package-lock.json...

File: lib/lroPollStrategy.ts [hunk 1]
```diff
@@ -102,7 +105,15 @@ export abstract class LROPollStrategy {
       this._status = getProvisioningState(result.parsedBody) || "Succeeded";
       this._response = result;
       this._mostRecentRequest = result.request;
-      this._resource = getResponseBody(result);
+      try {
+        this._resource = getResponseBody(result);
+      } catch (error) {
+        this._resource = undefined;
+        const resultStatus: number = result.status;
+        if (this._initialRequestMethod !== "DELETE" || resultStatus < 400 || 499 < resultStatus) {
```
Comment (RikkiGibson) [inline lib/lroPollStrategy.ts]: So is the idea that getResponseBody is allowed to fail, but only when either:

- the original request method was a DELETE
- the response status (for the original request? the most recent request?) indicates a request error

File: lib/lroPollStrategy.ts [hunk 2]
```diff
@@ -335,7 +349,17 @@ class AzureAsyncOperationLROPollStrategy extends LROPollStrategy {
    */
   protected sendPollRequest(): Promise<void> {
     return this.getStatus(this._azureAsyncOperationHeaderValue!).then((response: HttpOperationResponse) => {
+      const statusCode: number = response.status;
       const parsedResponse: any = response.parsedBody;
+      if (statusCode !== 200 && statusCode !== 201 && statusCode !== 202 && statusCode !== 204) {
```
Comment (RikkiGibson) [inline lib/lroPollStrategy.ts]: Should any of these status code checks ever vary by the expected responses present in the OperationSpec?

File: lib/lroPollStrategy.ts (hunk 1 above)
Comment (daschult) [reply lib/lroPollStrategy.ts]: Almost. It's not an either (or), it's an and. If the initial request was a DELETE and the result status is in the 400s, then we throw.

File: lib/lroPollStrategy.ts (hunk 2 above)
Comment (daschult) [reply lib/lroPollStrategy.ts]: If we ever get to specifying LRO polling status codes in the OperationSpec/Swagger, then we can revisit this, but I don't believe we do that now.

File: lib/lroPollStrategy.ts [hunk 3]
```diff
@@ -400,6 +424,17 @@ class AzureAsyncOperationLROPollStrategy extends LROPollStrategy {
 class GetResourceLROPollStrategy extends LROPollStrategy {
   protected sendPollRequest(): Promise<void> {
     return this.getStatus(this._initialRequestUrl).then(result => {
+      const statusCode: number = result.status;
+      const responseBody: any = result.parsedBody;
+      if (statusCode !== 200 && statusCode !== 201 && statusCode !== 202 && statusCode !== 204) {
+        const error = new RestError(`Invalid status code with response body "${result.bodyAsText}" occurred when polling for operation status.`);
+        error.statusCode = statusCode;
+        error.request = stripRequest(result.request);
+        error.response = result;
```
Comment (RikkiGibson) [inline lib/lroPollStrategy.ts]: Is there stripping that should happen on the response as well?

Comment (daschult) [reply lib/lroPollStrategy.ts]: I've thought about this previously, but currently we don't strip the response anywhere. If we find headers that should be removed, then we should add a stripResponse() method, but we don't have one now.

File: lib/lroPollStrategy.ts (hunk 1 above)
Comment (RikkiGibson) [reply lib/lroPollStrategy.ts]: I'm struggling to reconcile the code with your description just by staring at them, but if the tests pass I'm happy. This code makes it seem like an initial `"GET"` or a `"POST"` will always result in a rethrow, for instance.

Comment (daschult) [reply lib/lroPollStrategy.ts]: Oh, I'm sorry. I'm an idiot. You're right. If it's a delete and in the 400s, then we don't want to throw. All other scenarios that produce an error while parsing the response body do want to throw. You were right.

**Pull request merged**